Okta/OktaAuth/OktaAppAuth.swift

```diff
@@ -10,98 +10,105 @@
  * See the License for the specific language governing permissions and limitations under the License.
  */
 
-// Current version of the SDK
-let VERSION = "2.1.0"
+import UIKit
 
-// Holds the browser session
-public var currentAuthorizationFlow: OIDExternalUserAgentSession?
+public class OktaAppAuth {
 
-// Cache Okta.plist for reference
-public var configuration = try? OktaAuthConfig.default()
+    // Current version of the SDK
+    static let VERSION = "2.1.0"
 
-// Cache the Discovery Metadata
-public var discoveredMetadata: [String: Any]?
+    // Cache Okta.plist for reference
+    public let configuration: OktaAuthConfig
 
-// Token manager
-public var tokenManager = OktaTokenManager.readFromSecureStorage() {
-    didSet {
-        tokenManager?.writeToSecureStorage()
+    // Token manager
+    public internal(set) var tokenManager = OktaTokenManager.readFromSecureStorage() {
```

Comment:
Just thinking out-loud, we probably even don't need this property. As a developer I will receive tokenManager via callback. So I will continue to work with tokenManager and I don't need OktaAppAuth object anymore. My main object that I care about is tokenManager. So once application restarts I have to do: `OktaTokenManager.readFromSecureStorage()` and continue work with tokenManager. So there are separation of concerns:
- OktaAppAuth class - implements browser singIn/signOut flows
- OktaTokenManager class - tokens related functionality and authentication to resource endpoints

Reply:
After app relaunch it doesn't make sense for me to instantiate OktaAppAuth object(and provide config object) just to get access to tokenManager. I would rather use OktaTokenManager class directly.
Regarding signOut function, I've already provided comment that we have to implement additional function that accepts OktaTokenManager as a parameter. Without this function developer will not be able to do signOut in multi-account applications

Presence of tokenManager property in that class doesn't break anything, it just looks odd for me...

Reply:
@IldarAbdullin-okta OktaAppAuth class takes configuration on its initialization. SignOut operation uses configuration values from it. Therefore, clients cannot just pass TokenManager as a parameter, they need to pass a configuration that corresponds to a tokenManager.

Reply:
Right, and I don't see any problems why I just can't instantiate OktaAppAuth with required config and pass TokenManager as a parameter to signOut function.

Just consider the following use case. Let's say I have the following application environment:
- account1 = config1/tokenManager1
- account2 = config2/tokenManager2
- account3 = config3/tokenManager3

How I can do sign out for the account2 in the new implementation?

Reply:
Reworked.